packages/cli/src/metrics/prometheus-metrics.service.ts

```diff
@@ -5,196 +5,200 @@ import promBundle from 'express-prom-bundle';
 		promClient.register.clear(); // clear all metrics in case we call this a second time
-		this.setupDefaultMetrics();
-		this.setupN8nVersionMetric();
-		this.setupCacheMetrics();
-		this.setupMessageEventBusMetrics();
-		this.setupApiMetrics(app);
+		this.initDefaultMetrics();
+		this.initN8nVersionMetric();
+		this.initCacheMetrics();
+		this.initEventBusMetrics();
+		this.initApiMetrics(app);
 		this.mountMetricsEndpoint(app);
 	}
 
-	private setupN8nVersionMetric() {
-		const n8nVersion = semverParse(N8N_VERSION || '0.0.0');
+	/**
+	 * Set up metric for n8n version: `n8n_version_info`
+	 */
+	private initN8nVersionMetric() {
+		const n8nVersion = semverParse(N8N_VERSION);
 
-		if (n8nVersion) {
-			const versionGauge = new promClient.Gauge({
-				name: config.getEnv('endpoints.metrics.prefix') + 'version_info',
-				help: 'n8n version info.',
-				labelNames: ['version', 'major', 'minor', 'patch'],
-			});
+		if (!n8nVersion) return;
 
-			versionGauge.set(
-				{
-					version: 'v' + n8nVersion.version,
-					major: n8nVersion.major,
-					minor: n8nVersion.minor,
-					patch: n8nVersion.patch,
-				},
-				1,
-			);
-		}
+		const versionGauge = new promClient.Gauge({
+			name: this.prefix + 'version_info',
+			help: 'n8n version info.',
+			labelNames: ['version', 'major', 'minor', 'patch'],
+		});
+
+		const { version, major, minor, patch } = n8nVersion;
+
+		versionGauge.labels({ version: 'v' + version, major, minor, patch });
+		versionGauge.set(1);
 	}
 
-	private setupDefaultMetrics() {
-		if (config.getEnv('endpoints.metrics.includeDefaultMetrics')) {
-			promClient.collectDefaultMetrics();
-		}
+	/**
+	 * Set up default metrics collection with `prom-client`, e.g.
+	 * `process_cpu_seconds_total`, `process_resident_memory_bytes`, etc.
+	 */
+	private initDefaultMetrics() {
+		if (!this.includes.metrics.default) return;
+
+		promClient.collectDefaultMetrics();
 	}
 
-	private setupApiMetrics(app: express.Application) {
-		if (config.getEnv('endpoints.metrics.includeApiEndpoints')) {
-			const metricsMiddleware = promBundle({
-				autoregister: false,
-				includeUp: false,
-				includePath: config.getEnv('endpoints.metrics.includeApiPathLabel'),
-				includeMethod: config.getEnv('endpoints.metrics.includeApiMethodLabel'),
-				includeStatusCode: config.getEnv('endpoints.metrics.includeApiStatusCodeLabel'),
-			});
+	/**
+	 * Set up metrics for API endpoints with `express-prom-bundle`
+	 */
+	private initApiMetrics(app: express.Application) {
+		if (!this.includes.metrics.api) return;
+
+		const metricsMiddleware = promBundle({
+			autoregister: false,
+			includeUp: false,
+			includePath: this.includes.labels.apiPath,
+			includeMethod: this.includes.labels.apiMethod,
+			includeStatusCode: this.includes.labels.apiStatusCode,
+		});
 
-			app.use(['/rest/', '/webhook/', 'webhook-test/', '/api/'], metricsMiddleware);
-		}
+		app.use(['/rest/', '/webhook/', '/webhook-test/', '/api/'], metricsMiddleware);
```

Comment:
Others to include? e.g. `/webhook-waiting/`, `/form-waiting` Or excluded on purpose?

Reply:
good idea to include these 👍